x/ccv/utils/utils.go

```diff
@@ -30,6 +31,16 @@ func AccumulateChanges(currentChanges, newChanges []abci.ValidatorUpdate) []abci
 	for _, update := range m {
 		out = append(out, update)
 	}
+
+	// The list of tendermint updates should hash the same across all consensus nodes
+	// that means it is necessary to sort for determinism.
+	sort.Slice(out, func(i, j int) bool {
```

Comment:
Shouldn't this only compare key strings when the powers are equal?

Something like:
```golang
sort.Slice(ss, func(i, j int) bool {
	if ss[i].Power > ss[j].Power {
		return true
	} else if ss[i].Power == ss[j].Power {
		return ss[i].PubKey.String() > ss[j].PubKey.String()
	} else {
		return false
	}
})
```

example https://go.dev/play/p/7yV3siEjUr8

Reply:
Ahh you are right!!

Reply:
```go
sort.Slice(out, func(i, j int) bool {
	if out[i].Power != out[j].Power {
		return out[i].Power > out[j].Power
	}
	return out[i].PubKey.String() > out[j].PubKey.String()
})
```

Reply:
Nice catch @MSalopek.

Reply:
Yea nice catch. I think the original sort would be deterministic and basically OK, but it would be pretty unintuitive to understand what order they are in